src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -2947,8 +2947,6 @@ CopyRelationRestrictionContext(RelationRestrictionContext *oldContext)
 
 		/* not copyable, but readonly */
 		newRestriction->plannerInfo = oldRestriction->plannerInfo;
-		newRestriction->prunedShardIntervalList =
```

Comment:
> Do we want to set this to NULL? 

Yes, we do. The reason is that we don't have a copy function for `shardInterval`  and this is really not required to copy for our purposes.

I'll move it to another commit while I refactor the PR